### PR TITLE
[perf] Update redux store with only the values that changed

### LINKED ISSUE #5567: Selectively update the redux store upon receiving the evaluations to prevent whole canvas from re-rendering.

### PR DESCRIPTION
**Pull request by SatishGandham**
## Description
Earlier we used to replace the entire evaluated value in the redux store, now we selectively update only what have changed.
This change prevents unnecessary re-renders of the widgets.

Fixes #5567 
## Type of change
- Performance optimisation
-
## How Has This Been Tested?
Profiled before and after. Only modified widget is re-rendered.
## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: selectively-update-redux-store 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 53.58 **(-0.01)** | 35.32 **(-0.01)** | 32.13 **(0.01)** | 54.14 **(0)**
 :red_circle: | app/client/src/reducers/evaluationReducers/treeReducer.ts | 36.84 **(-48.87)** | 0 **(-100)** | 50 **(0)** | 50 **(-35.71)**
 :green_circle: | app/client/src/sagas/EvaluationsSaga.ts | 17.11 **(0.08)** | 1.01 **(0.02)** | 3.23 **(0.1)** | 20 **(0.2)**
 :red_circle: | app/client/src/workers/DataTreeEvaluator.ts | 88.7 **(0.2)** | 78.01 **(-0.16)** | 87.14 **(-0.18)** | 88.57 **(-0.1)**
 :green_circle: | app/client/src/workers/evaluationUtils.ts | 86.89 **(-0.07)** | 71.97 **(0.2)** | 87.5 **(0.27)** | 86.73 **(0.49)**</details>

### CODE REVIEW

File: app/client/src/workers/evaluation.worker.ts [hunk 1]
```diff
@@ -49,26 +50,22 @@ ctx.addEventListener(
...
+            // If functions exist, it will crash the web worker
+            dataTree = dataTree && JSON.parse(JSON.stringify(dataTree));
           } else {
-            dataTreeObject = dataTreeEvaluator.updateDataTree(unevalTree);
-            dataTree = dataTreeObject.dataTree;
-            evaluationOrder = dataTreeObject.evaluationOrder;
-            removedPaths = dataTreeObject.removedPaths;
+            dataTree = {};
+            const updateResponse = dataTreeEvaluator.updateDataTree(unevalTree);
+            updates = JSON.parse(JSON.stringify(updateResponse.updates));
```
Comment (SatishGandham) [inline app/client/src/workers/evaluation.worker.ts:66]: Is this to remove any functions?

**Review by somangshu — APPROVED** [review]
Approving to run tests

File: app/client/src/sagas/EvaluationsSaga.ts [hunk 2]
```diff
@@ -355,6 +357,7 @@ function* evaluateTreeSaga(
 ) {
   const unevalTree = yield select(getUnevaluatedDataTree);
   log.debug({ unevalTree });
+  console.log(unevalTree);
```
Comment (hetunandu) [inline app/client/src/sagas/EvaluationsSaga.ts:360]: remove console.log

**Review by hetunandu — APPROVED** [review]

**Review by somangshu — APPROVED** [review]
Approving to tun tests

**Review by riodeuno — APPROVED** [review]
Approving for tests

**Pull request merged**